rust/protocol/benches/session.rs

```diff
@@ -1,86 +1,125 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
 use criterion::{criterion_group, criterion_main, Criterion};
+use futures::executor::block_on;
 use libsignal_protocol_rust::*;
 
 #[path = "../tests/support/mod.rs"]
 mod support;
 
 pub fn session_encrypt_result(c: &mut Criterion) -> Result<(), SignalProtocolError> {
-    let (alice_session, bob_session) = support::initialize_sessions_v3()?;
-    let alice_session_record = SessionRecord::new(alice_session);
-    let bob_session_record = SessionRecord::new(bob_session);
+    let (alice_session_record, bob_session_record) = support::initialize_sessions_v3()?;
 
     let alice_address = ProtocolAddress::new("+14159999999".to_owned(), 1);
     let bob_address = ProtocolAddress::new("+14158888888".to_owned(), 1);
 
     let mut alice_store = support::test_in_memory_protocol_store();
     let mut bob_store = support::test_in_memory_protocol_store();
 
-    alice_store.store_session(&bob_address, &alice_session_record)?;
-    bob_store.store_session(&alice_address, &bob_session_record)?;
+    block_on(alice_store.store_session(&bob_address, &alice_session_record, None))?;
+    block_on(bob_store.store_session(&alice_address, &bob_session_record, None))?;
```

Comment:
Since it's a benchmark I wonder if it's better to use `expect_ready` than `block_on`, the latter being a little heavier. We'd have to move it down in the stack though.